posix/include/rtos/alloc.h

```diff
@@ -32,22 +32,28 @@
  *  @{
  */
 
+/*
+ * for compatibility with the initial `flags` meaning
+ * SOF_MEM_FLAG_ should start at BIT(2)
+ * the first two positions are reserved for SOF_BUF_ flags
+ */
+
```

Comment:
ok, looks like we need to split out the caps for buffers a tad more. 
So @jsarha is working on removing ```rballoc()``` to align with a single low level allocator and we need some way for buffer API to handle.
@dbaluta btw - I dont think we have any users of the ```SOF_BUF_OVERRUN_PERMITTED``` and ```SOF_BUF_UNDERRUN_PERMITTED``` - at least from grepping the topologies - IIRC these were a legacy flag around DMA based scheduling of pipelines (where > 1 DMAs could trigger at the same time). I think we could remove.

Reply:
@lgirdwood  I think this is a slightly better PR: https://github.com/thesofproject/sof/pull/10228

with less invasive modification.